src/test/stepFunctions/commands/visualizeStateMachine.test.ts

```diff
@@ -0,0 +1,464 @@
+            assert.ok(error instanceof Error)
+            const errorCasted = <Error>error
+            if (errorCasted.message === 'Could not get active text editor for state machine render.') {
+                assert.fail('Should not throw error on valid visualization')
+            } else {
+                assert.fail(errorCasted.message)
+            }
+        }
+    })
+
+    it('Test AslVisualizationManager managedVisualizations set does not add duplicate renders when multiple Vis active', async () => {
+        assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 0)
+
+        try {
+            // Preview Doc1
+            mockVsCode.showTextDocument(mockTextDocumentOne)
+            await aslVisualizationManager.visualizeStateMachine(mockGlobalStorage)
+            assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 1)
+
+            // Preview Doc2
+            mockVsCode.showTextDocument(mockTextDocumentTwo)
+            await aslVisualizationManager.visualizeStateMachine(mockGlobalStorage)
+            assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 2)
+
+            // Preview Doc1 Again
+            mockVsCode.showTextDocument(mockTextDocumentOne)
+            await aslVisualizationManager.visualizeStateMachine(mockGlobalStorage)
+            assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 2)
+
+            // Preview Doc2 Again
+            mockVsCode.showTextDocument(mockTextDocumentTwo)
+            await aslVisualizationManager.visualizeStateMachine(mockGlobalStorage)
+            assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 2)
+
+            const managedVisualizations = aslVisualizationManager.getManagedVisualizations()
+            assert.ok(managedVisualizations.get(mockTextDocumentOne.uri.path))
+            assert.ok(managedVisualizations.get(mockTextDocumentTwo.uri.path))
+        } catch (error) {
+            assert.ok(error instanceof Error)
+            const errorCasted = <Error>error
+            if (errorCasted.message === 'Could not get active text editor for state machine render.') {
+                assert.fail('Should not throw error on valid visualization')
+            } else {
+                assert.fail(errorCasted.message)
+            }
+        }
+    })
+
+    it('Test AslVisualizationManager managedVisualizations set removes visualization on visualization dispose, single vis', async () => {
+        assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 0)
+
+        try {
+            // Preview Doc1
+            mockVsCode.showTextDocument(mockTextDocumentOne)
+            const panel = await aslVisualizationManager.visualizeStateMachine(mockGlobalStorage)
+            assert.strictEqual(aslVisualizationManager.getManagedVisualizations().size, 1)
+
+            // Dispose of visualization panel
+            if (!panel) {
```

Comment:
you can use `assert.ok(panel, 'your fail message here')`

Reply:
Without this check there is an error: "Object is possibly undefined". I guess it doesn't detect the assert.ok as a check for undefined.